Fix a number of small zvol-related udev error handling issues

jgottula opened the pull request:
I've come across a number of issues of general "crustiness" concerning the udev rules file for zvol dev nodes, as well as the udev helper binary `zvol_id`. This PR consists of a small stack of patches that address some of that crustiness.

Some of these are very much "1 character of code changed, with 1 long novel of commit message explanation accompanying it".

### Motivation and Context

As mentioned in #12301, I noticed that when `zvol_id` runs into actual problems (which is not a common thing as far as I know), the way that errors are actually handled is... well... not good. The udev rules file for zvols is ancient and uses deprecated syntax, and the `zvol_id` helper binary is also ancient and has a number of interesting quicks itself.

The core problem in #12301 isn't something I can solve myself, at least right now. But the ancillary stuff, regarding old crusty udev rules files and poor error handling that tends to result in tons of completely spurious symlinks being created... well, that's something I would probably write some local patches for myself anyway; so why not turn it into a day-long writing exercise and contribute it back. 😝

### Description

`60-zvol.rules.in`
- Fix deprecated syntax: no commas between key-value pairs
- Fix deprecated syntax: use `$devnode` rather than `$tempnode`
- Ensure that the `PROGRAM` key definitely uses the match (`==`) operator so that success/failure of the program is actually checked
  - I believe my suspicions here ended up being somewhat of a red herring: pretty sure that actually it was a circumstance of "program is returning 'nonzero' exit status 512, which gets modulo'd to 0"; but being more explicit and avoiding mushy udev policies about "auto-correcting" technically-wrong operators into correct ones isn't a bad thing

`zvol_id_main.c`
- Alleviate Dennis Ritchie continually rolling in his grave due to having literally invented `stderr`, only to have its existence be completely disregarded by this program 👻
- Make the overall exit status logic less stupid, less problematic, and just generally easier to follow

### How Has This Been Tested?

- [x] I've tested the udev rule changes myself already on my test system, with udev `log_level=debug` etc
- [ ] I'll be testing the `zvol_id` changes in a matter of minutes, most likely, once I give my system a reboot

Fortunately both the rules file and the helper program are pretty well isolated in their own corner with udev and friends, and they don't really have any super duper strong interactions with other stuff; at least not in the way that changes to e.g. parts of the kernel module might.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
  - Well... only insofar as the spurious creation of incorrect symlinks was making systemd-udevd take forever to finish device initialization because a bunch of different zvol devices would all try to claim ownership of symlinks `/dev/zvol/Unable` et al 😬
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

Not sure if any tests are super-warranted here. I did do a cursory inspection to see if there were any preexisting tests for e.g. `vdev_id` or something similar like that which I could crib from; but didn't seem to find much. 🤷

behlendorf reviewed (approved):
Thanks for taking the time to remove this cruft and submitting those changes back.  This looks great, the detailed commit messages were very welcome since some of these changes would otherwise be non obvious!

jgottula commented:
Force-pushed just now:
- fixed a small parenthesis typo in the ~~description~~ essay part of the commit message for "Print zvol_id error messages to stderr rather than stdout"
- rebased onto the most latestest master commit (was going to force-push anyway, so why not)

pzakha commented:
Great write-up! The commit messages are definitely helpful and your analysis in https://github.com/openzfs/zfs/issues/12301 was very informative. Thanks for doing this work!

Regarding the removal of compatibility "/dev/pool/" links, I think it's a good idea. It might break some user configurations but we need to stop supporting deprecated formats at some point.
